freshmaker/pyxis.py

```diff
@@ -215,6 +215,121 @@ def get_images_by_digests(self, digests):
+            auto_rebuild tags of that repo
+        """
+        if not reg_repo_info:
+            return None
+        fltr = ""
+        # Construct filter for future request to Pyxis with registry-repository pairs
+        for reg, repo in reg_repo_info.keys():
+            if not fltr:
+                fltr = f"(registry=={reg};repository=={repo})"
+            else:
+                fltr += f",(registry=={reg};repository=={repo})"
+        params = {'include': ','.join(['data.auto_rebuild_tags',
+                                       'data.registry', 'data.repository']),
+                  'filter': fltr}
+        repos = self._pagination('repositories', params)
+
+        # For every repo add it's auto_rebuild_tags info
+        for repo in repos:
+            reg_repo_pair = (repo['registry'], repo['repository'])
+            # one of repos isn't in previously constructed map,
+            # so there is inconsistency
+            if reg_repo_pair not in reg_repo_info:
+                log.warning("There is inconsistency in naming for: %s/%s",
+                            reg_repo_pair[0], reg_repo_pair[1])
+                continue
+            tags = repo.get('auto_rebuild_tags')
+            # If the repository doesn't have 'auto_rebuild_tags', don't proceed with it
+            if tags:
+                reg_repo_info[reg_repo_pair]['auto_rebuild_tags'] = set(tags)
+            else:
+                del reg_repo_info[reg_repo_pair]
+
+    def _filter_auto_rebuild_nvrs(self, reg_repo_info):
+        """
+        For every pair in mapping get history of every auto_rebuild tag
+        filtering all nvrs from that registry/repo pair.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        :return: NVRs that were once tagged with an autorebuild tag
+        :rtype set(str):
+        """
+        params = {'include': 'data.brew.build'}
+        ret_nvrs = set()
+        for reg_repo_pair, info in reg_repo_info.items():
+            if not info.get('nvrs'):
+                continue
+            fltr = f"brew.build=in=({','.join(info['nvrs'])})"
+            params["filter"] = fltr
+            reg, repo = reg_repo_pair
+            for tag in info.get('auto_rebuild_tags', []):
+                tag_history = \
+                    self._pagination(f"repositories/registry/{reg}/"
+                                     f"repository/{repo}/tag/{tag}", params)
+                # If there is at least one record with nvr(brew.build) of the
+                # image, it means that image was published to a repo with
+                # auto_rebuild tag
+                if tag_history:
+                    auto_rebuild_nvrs = \
+                        [temp_tag['brew']['build'] for temp_tag in tag_history]
+                    ret_nvrs.update(auto_rebuild_nvrs)
```

Comment:
This is easier to read, more memory efficient, and should be slightly faster. For future reference, you can define a set with the list comprehension syntax by using curly braces (e.g. `{temp_tag['brew']['build'] for temp_tag in tag_history}`).
```suggestion
                    for temp_tag in tag_history:
                        ret_nvrs.add(temp_tag['brew']['build'])
```